### PR TITLE
React phone number input

### PR DESCRIPTION
**Pull request by AdrienEtienne**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/react-phone-number-input/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    onChange: (value: string) => void;
+    value: string;
+    // Optional props
+    onFocus?: () => void;
+    onBlur?: () => void;
+    country?: string;
+    metadata?: object;
+}
+
+export interface PhoneInputProps {
```
Comment (fleck) [inline types/react-phone-number-input/index.d.ts]: from the docs "See the list of all available props for <PhoneInput/>. All properties not listed there will be passed through to the phone number <input/> component." 
The props for this should also take any valid `<input/>` props.

```typescript
export interface PhoneInputProps extends Omit<React.InputHTMLAttributes<HTMLInputElement>, "onChange" | "id"> {
```

Comment (AdrienEtienne) [reply types/react-phone-number-input/index.d.ts]: Thanks.
Have created a PR for that: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/41898

**Comment by JeremyKirkham** [comment]
@AdrienEtienne  - also missing `numberInputProps` optional property.

**Comment by AdrienEtienne** [comment]
> @AdrienEtienne - also missing `numberInputProps` optional property.

Thank you for your comment.
See this PR should fix that: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/41957

**Pull request merged**